openmc/tallies.py

```diff
@@ -1611,6 +1635,15 @@ def _align_tally_data(self, other):
         ----------
         other : Tally
             The tally to outer product with this tally
+        filter_product : str
```

Comment:
Ditto my comment above - use `{'tensor', 'entrywise'}` here for the accepted types for `filter_product`, `nuclide_product` and `score_product`.